olp-cpp-sdk-dataservice-read/src/VersionedLayerClientImpl.cpp

```diff
@@ -305,6 +305,11 @@ client::CancellationToken VersionedLayerClientImpl::PrefetchTiles(
           futures->emplace_back(promise->get_future());
           auto context_it = contexts.emplace(contexts.end());
 
+          // Settings structure consumes a noticeable number of bytes in the
+          // task, shared pointer saves a lot of runtime memory.
+          auto shared_settings =
+              std::make_shared<client::OlpClientSettings>(settings);
```

Comment:
`settings` is just a copy of client member `settings_`. Maybe, it makes sense to create shared_settings from it at the begging of the PrefetchTiles().

Reply:
We try to achieve stateless client, in case we capture **this**, later on they might be changed in runtime, it will break the existing functionality.

Reply:
I mean, you are making second copy of client's settings. I suggest not capturing `this`, but remove variable `settings` (not `settings_`)at all and use shared_settings instead.
Do we actually need 2 copies?Am i missing something?